src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -971,8 +962,11 @@ VideoDecoder::DecodedOutput VideoDecoder::getFramePlayedAtTimestampNoDemux(
         return seconds >= frameStartTime && seconds < frameEndTime;
       });
   // Convert the frame to tensor.
-  auto output = convertAVFrameToDecodedOutput(rawOutput);
-  output.frame = MaybePermuteHWC2CHW(output.streamIndex, output.frame);
+  auto streamIndex = rawOutput.streamIndex;
+  auto preAllocatedOutputTensor = allocateEmptyHWCTensorForStream(streamIndex);
```

Comment:
In the light of this: https://github.com/pytorch/torchcodec/issues/312, it would be more correct to use rawOutput.frame's dimensions, and not the stream's dimensions